### PR TITLE
Trials part 2 - review

### PR DESCRIPTION
**Pull request by amanjeev**
I am committing stuff as I see. Please feel free to ignore.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,16 +82,19 @@ To actually enable the unsizing coercion of objects, the `CoerceUnsized` trait h
...
-### A non-delegating `CoerceUnsized` impl
+### 1. A non-delegating `CoerceUnsized` impl
 
 Such an impl is used for actual pointer like types, such as `&'a T` or `Arc<T>`.
-The implementing type and the `CoerceUnsized` target type must differ in a single generic parameter only.
-The differing generic parameters are required to then be part of a `T: Unsize<U>` bound where `T` is the relevant generic parameter of the implementing type and `U` the relevant generic parameter of the `CoerceUnsized` target type.
+The implementing type and the `CoerceUnsized` target type must differ in a single generic parameter only. Say, the parameters are `T` and `U`. Then,
+
+- `T` is the generic paramter of the implementing type; is bound as `T: Unsize<U>`
+- `U` is the generic paramter of the `CorecedUnsized` target type
```
Comment (bjorn3) [inline README.md]:
```suggestion
- `U` is the generic parameter of the `CoerceUnsized` target type
```

File: README.md [hunk 2]
```diff
@@ -80,16 +82,19 @@ To actually enable the unsizing coercion of objects, the `CoerceUnsized` trait h
...
 
-### A non-delegating `CoerceUnsized` impl
+### 1. A non-delegating `CoerceUnsized` impl
 
 Such an impl is used for actual pointer like types, such as `&'a T` or `Arc<T>`.
-The implementing type and the `CoerceUnsized` target type must differ in a single generic parameter only.
-The differing generic parameters are required to then be part of a `T: Unsize<U>` bound where `T` is the relevant generic parameter of the implementing type and `U` the relevant generic parameter of the `CoerceUnsized` target type.
+The implementing type and the `CoerceUnsized` target type must differ in a single generic parameter only. Say, the parameters are `T` and `U`. Then,
+
+- `T` is the generic paramter of the implementing type; is bound as `T: Unsize<U>`
```
Comment (bjorn3) [inline README.md]:
```suggestion
- `T` is the generic parameter of the implementing type; is bound as `T: Unsize<U>`
```

File: README.md [hunk 3]
```diff
@@ -230,21 +243,24 @@ The implementation will be as follows (and live in `core`):
...
+// - `[T; N]` is a contiguous slice of `T`'s, so a pointer pointing to its data is valid to
+//   be interpreted as a pointer to a slice `[T]`.
 unsafe impl<T, const N: usize> Unsize<[T]> for [T; N] {
     fn target_metadata((): <Self as Pointee>::Metadata) -> <[T] as Pointee>::Metadata {
         N
     }
 }
 ```
 
+### `CoercedUnsized`
```
Comment (bjorn3) [inline README.md]:
```suggestion
### `CoerceUnsized`
```

File: README.md [hunk 4]
```diff
@@ -230,23 +243,28 @@ The implementation will be as follows (and live in `core`):
...
 For the delegating implementations, the implementation of the `fn coerce_unsized` function will merely delegate to the inner value and then wrap that result again.
 
 ## Implementations provided by the compiler
 
-> Note: This section uses fictional rust syntax
+> ⚠️ Note: This section uses fictional rust syntax
 
 The compiler will generate `Unsize` implementations for types to trait object for their implemented types as before:
+
+For types to trait object for their implemented types, the compiler will generate `Unsize` implmentations:
```
Comment (Veykril) [inline README.md:267]: Now this is duplicated

File: README.md [hunk 5]
```diff
@@ -346,30 +366,32 @@ where
...
+        // so we the pinned contract is kept up
         unsafe { Pin::new_unchecked(self.pointer.coerce_unsized()) }
     }
 }
 ```
 
 
 While this is a breaking change, it should be in line with being a soundness fix.
-Unfortunately, these kind of impl requires negative bounds and negative reasoning which is its own can of worms and therefor likely not to happen, see https://github.com/rust-lang/rust/issues/42721.
+Unfortunately, these kind of impl requires negative bounds and negative reasoning which is its own can of worms and therefore likely not to happen, see [GH issue: Need negative trait bound #42721](https://github.com/rust-lang/rust/issues/42721).
```
Comment (Veykril) [inline README.md:387]: I always get this wrong 🤦‍♂️

**Review by Veykril — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (amanjeev) [reply README.md:267]: jesus i made this mistake

**Pull request merged**